### PR TITLE
Bug 1727022: set upgradeable operator condition

### PR DESCRIPTION
**Pull request by ericavonb**
Set the condition "upgradeable" on the "machine-config"  cluster operator.
https://github.com/openshift/api/blob/9525304a0adb725ab4a4a54539a1a6bf6cc343d3/config/v1/types_cluster_operator.go#L140

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1727022

### CODE REVIEW

File: pkg/operator/status.go [hunk 1]
```diff
@@ -207,6 +207,7 @@ func (optr *Operator) initializeClusterOperator() (*configv1.ClusterOperator, er
 	cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorAvailable, Status: configv1.ConditionFalse})
 	cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorProgressing, Status: configv1.ConditionFalse})
 	cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorDegraded, Status: configv1.ConditionFalse})
+	cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorUpgradeable, Status: configv1.ConditionTrue})
```
Comment (ericavonb) [inline pkg/operator/status.go]: WIP - this is NOT where I plan on setting this condition in the final version

File: pkg/operator/status.go [hunk 2]
```diff
@@ -125,10 +125,6 @@ func (optr *Operator) syncProgressingStatus() error {
 }
 
 func (optr *Operator) updateStatus(co *configv1.ClusterOperator, status configv1.ClusterOperatorStatusCondition) error {
-	existingCondition := cov1helpers.FindStatusCondition(co.Status.Conditions, status.Type)
-	if existingCondition.Status != status.Status {
-		status.LastTransitionTime = metav1.Now()
-	}
```
Comment (runcom) [inline pkg/operator/status.go]: any reason to have dropped this?

Comment (ericavonb) [reply pkg/operator/status.go]: It was causing a nil-pointer dereference panic when the existing condition wasn't found for the new type, but the logic is already handled in the call to `cov1helpers.SetStatusCondition` anyways so I just removed it.

https://github.com/openshift/library-go/blob/3679c350479175dceb60cbdafe68f0d92280952f/pkg/config/clusteroperator/v1helpers/status.go#L31

Comment (runcom) [reply pkg/operator/status.go]: 👍👍

**Pull request merged**